### PR TITLE
new lesson on object interactions

### PR DESCRIPTION
**Pull request by linda-seiter**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,31 +2,435 @@
...
+## Code Along
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+Fork and clone this lesson.  We will build an application
+that demonstrates interactions among associated objects:
 
-## Topic 2
+- a `Motorcycle` object is associated with two `Tire` objects
```
Comment (kcleland0818) [inline README.md]: Nit: capitalize the "A"s?

File: README.md [hunk 2]
```diff
@@ -2,31 +2,435 @@
...
-appropriate to help with the reader's orientation with the material and future
-referencing.
+- The `Main` class contains a `main` method that creates an instance of `Motorcycle`
+  and implicitly calls the `toString` and explicitly calls the`rideThroughMud` methods on the object.
+- The `Motorcycle` class has fields for the `make` and `model`, along
+  with  fields `frontTire` and `rearTire` that reference instances of the `Tire` class.
+- The `Car` class has fields for the `make` and `model`, along
+  with an array named `tires` that stores references to 4 `Tire` objects.
+- The `Tire` class has instance variables `airPressure` and `clean`,
+  along with class constants `MIN_RECOMMENDED_PRESSURE` and `MIN_RECOMMENDED_PRESSURE`.
```
Comment (kcleland0818) [inline README.md]: I think one of these should be `MAX_RECOMMENDED_PRESSURE`

File: README.md [hunk 3]
```diff
@@ -2,31 +2,435 @@
...
+    System.out.println("Before checking the tires");
+    System.out.println(myHonda);
+    myHonda.checkTires();
+    System.out.println("After checking the tires");
+    System.out.println(myHonda);
+}
+```
+
+All car tires are initially low (your values may differ), but
+after checking the tires they should all have the maximum air pressure:
```
Comment (kcleland0818) [inline README.md]: Typo: I think "all" should be "call"

**Review by kcleland0818 — COMMENTED** [review]
A couple of typos

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md]: reworded

**Comment by linda-seiter** [comment]
added message numbers in diagrams and code

**Review by developerjaym — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -2,31 +2,483 @@
...
+![breakpoint rideThroughMud](https://curriculum-content.s3.amazonaws.com/6676/java-multipleclasses/breakpoint_ridethroughmud.png)
+
+- Use 'Step Over' for the print statements.
+- Use 'Step Into' for the `Motorcycle` and `Tire` method calls.
+
+### Message #1 `myHarley.rideThroughMud()`
+
+Press the `debug` button to launch the debugger and stop at the breakpoint,
+then switch to the Java Visualizer view (the yellow arrow is added to show the
+current line of execution).  The `MotorCycle` object
```
Comment (kcleland0818) [inline README.md]: Lowercase c in `MotorCycle` to match the class name

**Review by kcleland0818 — COMMENTED** [review]
Thanks for adding that conclusion with the final sequence diagram. I think that will help connect the dots for students!

File: README.md (hunk 4 above)
Comment (linda-seiter) [reply README.md]: wow, good catch.

**Review by kcleland0818 — APPROVED** [review]

**Review by AlveeM — APPROVED** [review]

**Pull request merged**